packages/react-core/src/demos/Button.md

```diff
@@ -0,0 +1,75 @@
+---
+id: Button
+section: components
+---
+
+import CheckCircleIcon from '@patternfly/react-icons/dist/esm/icons/check-circle-icon';
+
+## Demos
+
+### Progress button
+
+```ts
+import React from 'react';
+import { Form, FormGroup, ActionGroup, InputGroup, TextInput, Button } from '@patternfly/react-core';
+import CheckCircleIcon from '@patternfly/react-icons/dist/esm/icons/check-circle-icon';
+
+const ProgressButton = () => {
+  const [loginState, setLoginState] = React.useState<'notLoggedIn' | 'loading' | 'loggedIn'>('notLoggedIn');
+  const actionButton = {
+    notLoggedIn: (
+      <Button
+        variant="primary"
+        onClick={() => {
+          setLoginState('loading');
+          setTimeout(() => {
+            setLoginState('loggedIn');
+          }, 3000);
+        }}
+      >
+        Link account and log in
+      </Button>
+    ),
+    loading: (
```

Comment:
Just my $0.02, I would probably expect there to be a single button with vars as values for the `isLoading` and `icon` props, and the button text. The `onClick` function would update those vars to update the props/text. Seems like a more DRY approach. Though the way you've done it here is valid, too.

Reply:
Sure, changed to be 1 button